rasterio/_env.pyx

```diff
@@ -68,11 +68,11 @@ cdef void log_error(CPLErr err_class, int err_no, const char* msg) with gil:
 # other platforms. Each calls log_error().
 IF UNAME_SYSNAME == "Windows":
     cdef void __stdcall logging_error_handler(CPLErr err_class, int err_no,
-                                              const char* msg):
+                                              const char* msg) with gil:
         log_error(err_class, err_no, msg)
 ELSE:
     cdef void logging_error_handler(CPLErr err_class, int err_no,
-                                    const char* msg):
+                                    const char* msg) with gil:
```

Comment:
Okay, thanks for the tip, I'll look into gdb more. I tried it out a bit last night and only managed to get an output that showed it was a SIGSEGV 11 but didn't tell me where it occurred. Knowing at least that is the right tool for the job, I'll dig in a little more.

In the meantime, it seems like you should go ahead and merge, since my problem seems to be distinct, if similar.